File: game - daily progress/daily-progress-game/generate-icon.js
const fs = require('fs');
const path = require('path');
const { createCanvas } = require('canvas');

// Check if canvas is installed
try {
  require.resolve('canvas');
} catch (e) {
  console.log('Canvas module not found. Installing...');
  const { execSync } = require('child_process');
  execSync('npm install canvas', { stdio: 'inherit' });
}

function createGardenIcon(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // Background gradient
  const gradient = ctx.createLinearGradient(0, 0, size, size);
  gradient.addColorStop(0, '#66bb6a');
  gradient.addColorStop(1, '#4caf50');
  ctx.fillStyle = gradient;
  ctx.fillRect(0, 0, size, size);

  // Draw soil at bottom
  ctx.fillStyle = '#8d6e63';
  ctx.fillRect(0, size * 0.7, size, size * 0.3);

  // Draw sun
  ctx.fillStyle = '#ffd54f';
  ctx.beginPath();
  ctx.arc(size * 0.15, size * 0.15, size * 0.08, 0, Math.PI * 2);
  ctx.fill();

  // Sun rays
  ctx.strokeStyle = '#ffd54f';
  ctx.lineWidth = size * 0.015;
  for (let i = 0; i < 8; i++) {
    const angle = (i * Math.PI) / 4;
    const x1 = size * 0.15 + Math.cos(angle) * size * 0.11;
    const y1 = size * 0.15 + Math.sin(angle) * size * 0.11;
    const x2 = size * 0.15 + Math.cos(angle) * size * 0.15;
    const y2 = size * 0.15 + Math.sin(angle) * size * 0.15;
    ctx.beginPath();
    ctx.moveTo(x1, y1);
    ctx.lineTo(x2, y2);
    ctx.stroke();
  }

  // Draw center flower
  const flowerX = size * 0.5;
  const flowerY = size * 0.5;

  // Stem
  ctx.strokeStyle = '#2e7d32';
  ctx.lineWidth = size * 0.02;
  ctx.beginPath();
  ctx.moveTo(flowerX, flowerY);
  ctx.lineTo(flowerX, size * 0.75);
  ctx.stroke();

  // Petals
  ctx.fillStyle = '#ee5a6f';
  for (let i = 0; i < 6; i++) {
    const angle = (i * Math.PI * 2) / 6;
    const petalX = flowerX + Math.cos(angle) * size * 0.08;
    const petalY = flowerY + Math.sin(angle) * size * 0.08;
    ctx.beginPath();
    ctx.arc(petalX, petalY, size * 0.05, 0, Math.PI * 2);
    ctx.fill();
  }

  // Flower center
  ctx.fillStyle = '#fff200';
  ctx.beginPath();
  ctx.arc(flowerX, flowerY, size * 0.04, 0, Math.PI * 2);
  ctx.fill();

  // Left plant
  const plant1X = size * 0.25;
  const plant1Y = size * 0.55;

  // Stem
  ctx.strokeStyle = '#2e7d32';
  ctx.beginPath();
  ctx.moveTo(plant1X, plant1Y);
  ctx.lineTo(plant1X, size * 0.75);
  ctx.stroke();

  // Small flower petals
  ctx.fillStyle = '#ff6b9d';
  for (let i = 0; i < 5; i++) {
    const angle = (i * Math.PI * 2) / 5 - Math.PI / 2;
    const petalX = plant1X + Math.cos(angle) * size * 0.06;
    const petalY = plant1Y + Math.sin(angle) * size * 0.06;
    ctx.beginPath();
    ctx.arc(petalX, petalY, size * 0.04, 0, Math.PI * 2);
    ctx.fill();
  }

  // Right plant (leaves)
  const plant3X = size * 0.75;
  const plant3Y = size * 0.6;

  // Stem
  ctx.strokeStyle = '#2e7d32';
  ctx.beginPath();
  ctx.moveTo(plant3X, plant3Y);
  ctx.lineTo(plant3X, size * 0.75);
  ctx.stroke();

  // Leaves
  ctx.fillStyle = '#4caf50';
  // Left leaf
  ctx.beginPath();
  ctx.ellipse(plant3X - size * 0.05, plant3Y, size * 0.06, size * 0.03, -Math.PI / 4, 0, Math.PI * 2);
  ctx.fill();
  // Right leaf
  ctx.beginPath();
  ctx.ellipse(plant3X + size * 0.05, plant3Y, size * 0.06, size * 0.03, Math.PI / 4, 0, Math.PI * 2);
  ctx.fill();

  return canvas;
}

// Create directories
const dirs = [
  'resources',
  'android/app/src/main/res/mipmap-mdpi',
  'android/app/src/main/res/mipmap-hdpi',
  'android/app/src/main/res/mipmap-xhdpi',
  'android/app/src/main/res/mipmap-xxhdpi',
  'android/app/src/main/res/mipmap-xxxhdpi'
];

dirs.forEach(dir => {
  if (!fs.existsSync(dir)) {
    fs.mkdirSync(dir, { recursive: true });
  }
});

// Generate icons
const icons = [
  { path: 'resources/icon.png', size: 1024 },
  { path: 'android/app/src/main/res/mipmap-mdpi/ic_launcher.png', size: 48 },
  { path: 'android/app/src/main/res/mipmap-hdpi/ic_launcher.png', size: 72 },
  { path: 'android/app/src/main/res/mipmap-xhdpi/ic_launcher.png', size: 96 },
  { path: 'android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png', size: 144 },
  { path: 'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png', size: 192 },
  { path: 'android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png', size: 48 },
  { path: 'android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png', size: 72 },
  { path: 'android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png', size: 96 },
  { path: 'android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png', size: 144 },
  { path: 'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png', size: 192 },
  { path: 'android/app/src/main/res/mipmap-mdpi/ic_launcher_foreground.png', size: 108 },
  { path: 'android/app/src/main/res/mipmap-hdpi/ic_launcher_foreground.png', size: 162 },
  { path: 'android/app/src/main/res/mipmap-xhdpi/ic_launcher_foreground.png', size: 216 },
  { path: 'android/app/src/main/res/mipmap-xxhdpi/ic_launcher_foreground.png', size: 324 },
  { path: 'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_foreground.png', size: 432 }
];

icons.forEach(({ path: iconPath, size }) => {
  const canvas = createGardenIcon(size);
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(iconPath, buffer);
  console.log(`Created: ${iconPath} (${size}x${size})`);
});

console.log('\nAll icons generated successfully!');
console.log('The app icon has been updated for Android.');

File: game - daily progress/daily-progress-game/generate_icon.py
from PIL import Image, ImageDraw
import os

def create_garden_icon(size):
    # Create a new image with transparent background
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Background gradient effect (green)
    for i in range(size):
        color_intensity = int(70 + (20 * i / size))  # Gradient from darker to lighter green
        color = (76, color_intensity + 100, 80, 255)  # Green gradient
        draw.rectangle([0, i, size, i+1], fill=color)
    
    # Draw soil at bottom
    soil_height = int(size * 0.3)
    draw.rectangle([0, size - soil_height, size, size], fill=(141, 110, 99, 255))
    
    # Draw sun
    sun_x, sun_y = int(size * 0.15), int(size * 0.15)
    sun_radius = int(size * 0.08)
    draw.ellipse([sun_x - sun_radius, sun_y - sun_radius, 
                  sun_x + sun_radius, sun_y + sun_radius], 
                 fill=(255, 213, 79, 255))
    
    # Draw center flower (main focus)
    flower_x = size // 2
    flower_y = int(size * 0.5)
    petal_size = int(size * 0.08)
    
    # Stem
    stem_width = max(2, int(size * 0.02))
    draw.rectangle([flower_x - stem_width//2, flower_y, 
                    flower_x + stem_width//2, int(size * 0.75)], 
                   fill=(46, 125, 50, 255))
    
    # Petals (pink/red)
    petal_color = (238, 90, 111, 255)
    for i in range(6):
        angle = i * 60
        import math
        px = flower_x + int(petal_size * math.cos(math.radians(angle)))
        py = flower_y + int(petal_size * math.sin(math.radians(angle)))
        draw.ellipse([px - petal_size//2, py - petal_size//2,
                      px + petal_size//2, py + petal_size//2],
                     fill=petal_color)
    
    # Flower center (yellow)
    center_size = int(size * 0.06)
    draw.ellipse([flower_x - center_size//2, flower_y - center_size//2,
                  flower_x + center_size//2, flower_y + center_size//2],
                 fill=(255, 202, 87, 255))
    
    # Left plant
    plant1_x = int(size * 0.25)
    plant1_y = int(size * 0.55)
    
    # Stem
    draw.rectangle([plant1_x - stem_width//2, plant1_y, 
                    plant1_x + stem_width//2, int(size * 0.75)], 
                   fill=(46, 125, 50, 255))
    
    # Small flower
    small_petal_size = int(size * 0.05)
    for i in range(5):
        angle = i * 72
        import math
        px = plant1_x + int(small_petal_size * 0.7 * math.cos(math.radians(angle)))
        py = plant1_y + int(small_petal_size * 0.7 * math.sin(math.radians(angle)))
        draw.ellipse([px - small_petal_size//2, py - small_petal_size//2,
                      px + small_petal_size//2, py + small_petal_size//2],
                     fill=(255, 107, 157, 255))
    
    # Right plant (leaves only)
    plant3_x = int(size * 0.75)
    plant3_y = int(size * 0.6)
    
    # Stem
    draw.rectangle([plant3_x - stem_width//2, plant3_y, 
                    plant3_x + stem_width//2, int(size * 0.75)], 
                   fill=(46, 125, 50, 255))
    
    # Leaves
    leaf_color = (76, 175, 80, 255)
    leaf_size = int(size * 0.06)
    # Left leaf
    draw.ellipse([plant3_x - leaf_size, plant3_y - leaf_size//2,
                  plant3_x, plant3_y + leaf_size//2],
                 fill=leaf_color)
    # Right leaf
    draw.ellipse([plant3_x, plant3_y - leaf_size//2,
                  plant3_x + leaf_size, plant3_y + leaf_size//2],
                 fill=leaf_color)
    
    return img

# Create main directories if they don't exist
os.makedirs('resources', exist_ok=True)
os.makedirs('android/app/src/main/res/mipmap-hdpi', exist_ok=True)
os.makedirs('android/app/src/main/res/mipmap-mdpi', exist_ok=True)
os.makedirs('android/app/src/main/res/mipmap-xhdpi', exist_ok=True)
os.makedirs('android/app/src/main/res/mipmap-xxhdpi', exist_ok=True)
os.makedirs('android/app/src/main/res/mipmap-xxxhdpi', exist_ok=True)

# Generate icons
sizes = {
    'resources/icon.png': 1024,
    'android/app/src/main/res/mipmap-mdpi/ic_launcher.png': 48,
    'android/app/src/main/res/mipmap-hdpi/ic_launcher.png': 72,
    'android/app/src/main/res/mipmap-xhdpi/ic_launcher.png': 96,
    'android/app/src/main/res/mipmap-xxhdpi/ic_launcher.png': 144,
    'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher.png': 192,
    'android/app/src/main/res/mipmap-mdpi/ic_launcher_round.png': 48,
    'android/app/src/main/res/mipmap-hdpi/ic_launcher_round.png': 72,
    'android/app/src/main/res/mipmap-xhdpi/ic_launcher_round.png': 96,
    'android/app/src/main/res/mipmap-xxhdpi/ic_launcher_round.png': 144,
    'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.png': 192,
}

for path, size in sizes.items():
    icon = create_garden_icon(size)
    icon.save(path, 'PNG')
    print(f"Created: {path} ({size}x{size})")

# Also create adaptive icons for Android
adaptive_sizes = {
    'android/app/src/main/res/mipmap-mdpi/ic_launcher_foreground.png': 108,
    'android/app/src/main/res/mipmap-hdpi/ic_launcher_foreground.png': 162,
    'android/app/src/main/res/mipmap-xhdpi/ic_launcher_foreground.png': 216,
    'android/app/src/main/res/mipmap-xxhdpi/ic_launcher_foreground.png': 324,
    'android/app/src/main/res/mipmap-xxxhdpi/ic_launcher_foreground.png': 432,
}

for path, size in adaptive_sizes.items():
    icon = create_garden_icon(size)
    icon.save(path, 'PNG')
    print(f"Created adaptive: {path} ({size}x{size})")

print("\nAll icons generated successfully!")
print("The app icon has been updated for Android.")